### ISSUE TITLE
Missing 's3_bucket_name' argument

### ISSUE THREAD

**Issue by t0yv0** [pulumi/pulumi-aws#2933]
### What happened?

Examples started failing with 

```
Missing 's3_bucket_name' argument
```

### Example

```python

import pulumi
from pulumi_aws import s3
import pulumi_awsx as awsx

trail = awsx.cloudtrail.Trail("awsx-py-cloudtrail", enable_logging=True)

pulumi.export('tr_id', trail.id)

```

### Output of `pulumi about`

```
CLI          
Version      3.90.0
Go Version   go1.21.3
Go Compiler  gc

Plugins
NAME    VERSION
aws     6.6.1
awsx    2.0.3
docker  4.4.4
python  unknown

Host     
OS       darwin
Version  14.0
Arch     x86_64

This project is written in python: executable='/usr/bin/python3' version='3.9.6'

Current Stack: t0yv0/aws-cloudtrail/dev

Found no resources associated with dev

Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/t0yv0
User           t0yv0
Organizations  t0yv0, pulumi
Token type     personal

Dependencies:
NAME         VERSION
pip          23.3.1
pulumi-awsx  2.0.3
setuptools   68.2.2
wheel        0.41.2

Pulumi locates its logs in /var/folders/gk/cchgxh512m72f_dmkcc3d09h0000gp/T/ by default
```

requirements.txt

```
pulumi>=3.0.0,<4.0.0
pulumi-aws==6.6.1
pulumi-awsx>=2.0.0
```

### Additional context

Possibly a recent regression per @thomas11 

```
The awsx python examples tests [started failing](https://github.com/pulumi/pulumi-awsx/actions/runs/6667175340/job/18120341257?pr=1128) reproducibly and the error messages were introduced in 6.6.1: “[Missing ‘s3_bucket_name’ argument](https://github.com/pulumi/pulumi-aws/commit/4c1b30033eff20e718769b19531d86677affac57#diff-f76e9f053b9314f38222d16c18f54cad69eccc0c418fd71ccbbbdf49932a13ecR99)” and “Missing ‘default_actions’ argument”. The timing lines up as well.
I tested by editing the requirements.txt dependency to <6.6.1 and [that run](https://github.com/pulumi/pulumi-awsx/actions/runs/6668045036/job/18122926335?pr=1128) half succeeded: one of the two tests succeeded, in the other one pulumi decided to install aws 6.6.1 on pulumi up - why? (@ Updating....[resource plugin aws-6.6.1] installing)
Anyways, I think this is pretty conclusive.
The failing program with the CloudTrail “Missing ‘s3_bucket_name’ argument” is a single line:
trail = awsx.cloudtrail.Trail("awsx-py-cloudtrail", enable_logging=True)
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by thomas11** [comment]
Simplest repro:

```python
from pulumi_aws import cloudtrail
trail = cloudtrail.Trail("awsx-py-cloudtrail", s3_bucket_name="bucket",)
```

requirements.txt, switching between 6.6.0 and 6.6.1:
```
pulumi>=3.0.0,<4.0.0
pulumi-aws==6.6.1
```

**Comment by t0yv0** [comment]
aws:cloudtrail/trail:Trail has a requiredInputs populated for s3BucketName


In _utilities.py
```python
class Module(pulumi.runtime.ResourceModule):
    def __init__(self, mod_info):
        super().__init__()
        self.mod_info = mod_info

    def version(self):
        return _version

    def construct(self, name: str, typ: str, urn: str) -> pulumi.Resource:
        class_name = self.mod_info['classes'].get(typ, None)

        if class_name is None:
            raise Exception(f"unknown resource type {typ}")

        TheClass = getattr(lazy_import(self.mod_info['fqn']), class_name)
        print('Constructing', name, class_name, self.mod_info['fqn'])
        return TheClass(name, pulumi.ResourceOptions(urn=urn))
```

    # Constructing awsx-py-cloudtrail Trail pulumi_aws.cloudtrail

    # NOTE there no arguments passed

    # Dropping to __init__:

```python
    def __init__(__self__, resource_name: str, *args, **kwargs):
        resource_args, opts = _utilities.get_resource_args_opts(TrailArgs, pulumi.ResourceOptions, *args, **kwargs)
        if resource_args is not None:
            __self__._internal_init(resource_name, opts, **resource_args.__dict__)
        else:
            kwargs = kwargs or {}
            def _setter(key, value):
                kwargs[key] = value
            import json
            print("_configure HERE, KWARGS=" + json.dumps(kwargs))
            TrailArgs._configure(_setter, **kwargs)
            __self__._internal_init(resource_name, *args, **kwargs)
```

    # _configure HERE, KWARGS={}

    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 1029, in do_register
    #     rpc.resolve_outputs(
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 1104, in resolve_outputs
    #     for key, value in deserialize_properties(outputs).items():
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 637, in deserialize_properties
    #     value = deserialize_property(v, keep_unknowns)
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 769, in deserialize_property
    #     props = deserialize_properties(value, keep_unknowns)
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 617, in deserialize_properties
    #     return deserialize_resource(props_struct, keep_unknowns)
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi/runtime/rpc.py", line 670, in deserialize_resource
    #     return cast("Resource", resource_module.construct(urn_name, typ, urn))
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi_aws/_utilities.py", line 211, in construct
    #     return TheClass(name, pulumi.ResourceOptions(urn=urn))
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi_aws/cloudtrail/trail.py", line 1101, in __init__
    #     TrailArgs._configure(_setter, **kwargs)
    #   File "/Users/t0yv0/tmp/2023/10/aws-cloudtrail/venv/lib/python3.9/site-packages/pulumi_aws/cloudtrail/trail.py", line 102, in _configure
    #     raise TypeError("Missing 's3_bucket_name' argument : " + str(len(kwargs)))
    # TypeError: Missing 's3_bucket_name' argument : 0

**Comment by t0yv0** [comment]
This makes me suspect the https://github.com/pulumi/pulumi/issues?q=is%3Aissue+label%3Ap1+is%3Aclosed+_configure sequence of _configure issues has one more. For Construct calls (MLCs) _configure eagerly checks for missing required arguments but they're not made available to it.

**Comment by t0yv0** [comment]
Trying to build AWS against downgraded pulumi/pkg and see if that reproduces.

**Comment by t0yv0** [comment]
* [TypeError: Missing 'capacity_constraints_max' argument](https://github.com/pulumi/pulumi-ec/issues/172): commented, opened {kind/bug, needs-triage}
* [Routine bridge updates causes tests to fail with missing "resource_name"](https://github.com/pulumi/pulumi-confluentcloud/issues/248): commented, opened {kind/bug, needs-triage}

**Comment by justinvp** [comment]
This looks like it's the same as https://github.com/pulumi/pulumi-awsx/issues/1131#issuecomment-1783725349

https://github.com/pulumi/pulumi/pull/14427 has reverted the change that introduced `_configure` in the first place, which was released in v3.91.1. Once `pulumi-aws` and `pulumi-awsx` are updated to use the v3.91.1 codegen and new versions are released, this shouldn't be an issue.

Probably need to make sure `pulumi-awsx` is referencing the fixed version of `pulumi-aws`.

**Comment by iwahbe** [comment]
Closed by https://github.com/pulumi/pulumi-aws/pull/2938.
